controllers/auth.controller.js

```diff
@@ -0,0 +1,108 @@
+const httpStatus = require('http-status');
+const User = require('../models/user.model');
+const sendResponse = require('../helpers/response');
+const PasswordReset = require('../models/passwordReset.model');
+const sendMail = require('../helpers/SendMail');
+const TokenDecoder = require('../helpers/TokenDecoder');
+const bcrypt = require('bcrypt');
+
+exports.forgotPassword = async (req, res) => {
+  const user = await User.getByEmail(req.body.email);
+  if (!user) {
+    return res.json(sendResponse(httpStatus.NOT_FOUND, 'User not found'));
+  }
+  const passwordReset = new PasswordReset({
+    userID: user._id,
+    email: req.body.email,
+    isAdmin: user.isAdmin
+  });
+  const passwordResetResult = await passwordReset.save();
+  if (!passwordResetResult || !passwordResetResult.email) {
```

Comment:
Sorry, I don't understand what is to be achieved here. Kindly put me through.

Reply:
Was checking for issues that may arise while trying to save the password reset request was saved to db.